app/models/book.rb

```diff
@@ -1,3 +1,17 @@
 class Book < ApplicationRecord
-  
+  validates_presence_of :title
+  has_many :reviews
+
+  def average_rating
+    ratings = reviews.map {|review| review.rating}
+    ratings.inject(:+) / reviews.count
+  end
+
+  def best_review
+    reviews.order("rating DESC").first
+  end
+
+  def worst_review
+    reviews.order("rating ASC").first
+  end
```

Comment:
I'm glad you used ActiveRecord for these two methods, but remember the aggregate/calculation methods you have available to you.

Reply:
I take that back, your aggregate would get you the number for the highest/lowest but not the object and I see you use these two methods to get the numbers and the objects.